### PR TITLE
SLING-7313 refresh the resource type resolver as well

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: src/main/java/org/apache/sling/resourceresolver/impl/helper/ResourceResolverControl.java [hunk 1]
```diff
@@ -148,12 +148,20 @@ private void logout() {
...
-     * Refreshes all refreshable providers.
+     * Refreshes all refreshable providers as well as the resolver used for resource types.
      */
     public void refresh(@Nonnull final ResourceResolverContext context) {
         for (final AuthenticatedResourceProvider p : context.getProviderManager().getAllUsedRefreshable()) {
             p.refresh();
         }
+        if (this.resourceTypeResourceResolver != null) {
+            try {
+                this.resourceTypeResourceResolver.close();
```
[the remark this answers is not in the document]
Comment (kwin) [reply src/main/java/org/apache/sling/resourceresolver/impl/helper/ResourceResolverControl.java]: Yes, of course. Good catch

File: src/main/java/org/apache/sling/resourceresolver/impl/helper/ResourceResolverControl.java [hunk 2]
```diff
@@ -148,12 +148,20 @@ private void logout() {
...
+            } catch ( final Throwable t) {
+                // the resolver (or the underlying provider) might already be terminated (bundle stopped etc.)
```
[the remark this answers is not in the document]
Comment (kwin) [reply src/main/java/org/apache/sling/resourceresolver/impl/helper/ResourceResolverControl.java]: Ok, agree.

**Comment by kwin** [comment]
@jsedding Please check again. The only thing I am not sure about is concurrency here. May this context be used from two different threads? If so isn't it dangerous to not synchronize close and refresh? After the check for null the context may have been closed in another thread... Then an NPE would be thrown...

**Comment by jsedding** [comment]
@kwin I don't think there's an issue with synchronization here. AFAIU we're in the implementation details of a single ResourceResolver, which is documented as NOT thread-safe. Therefore it would be a programming error if it was used concurrently.

cc @cziegeler you know these parts well, I would appreciate if you could double-check. Thanks.

**Pull request merged**